File: StayBooking/StayBooking/src/main/java/com/dto/CreateBookingRequestDto.java
package com.dto;

import jakarta.validation.constraints.NotNull;
import jakarta.validation.constraints.Positive;
import lombok.Data;

import java.time.LocalDate;

@Data
public class CreateBookingRequestDto {
    @NotNull(message = "userId is required")
    private Integer userId;

    @NotNull(message = "hotelId is required")
    private Integer hotelId;

    @NotNull(message = "roomId is required")
    private Integer roomId;

    @NotNull(message = "checkInDate is required")
    private LocalDate checkInDate;

    @NotNull(message = "checkOutDate is required")
    private LocalDate checkOutDate;

    @NotNull(message = "numberOfGuests is required")
    @Positive(message = "numberOfGuests must be greater than 0")
    private Integer numberOfGuests;
}

File: StayBookingFrontend/lib/model/create_booking_request_dto.dart
class CreateBookingRequestDto {
  const CreateBookingRequestDto({
    required this.userId,
    required this.hotelId,
    required this.roomId,
    required this.checkInDate,
    required this.checkOutDate,
    required this.numberOfGuests,
  });

  final int userId;
  final int hotelId;
  final int roomId;
  final String checkInDate;
  final String checkOutDate;
  final int numberOfGuests;

  Map<String, dynamic> toJson() {
    return {
      'userId': userId,
      'hotelId': hotelId,
      'roomId': roomId,
      'checkInDate': checkInDate,
      'checkOutDate': checkOutDate,
      'numberOfGuests': numberOfGuests,
    };
  }
}
